src/drawers/drawer.ts

```diff
@@ -76,7 +76,14 @@ export class Drawer {
    * @param{any[]} data The data to be drawn
    */
   private _bindSelectionData(data: any[]) {
-    let dataElements = this.selection().data(data);
+    // if the dataset has a key, use it when binding the data   
+    let dataElements: d3.selection.Update<any>;
+    if (this._dataset && this._dataset.key()) {
+      dataElements = this.selection().data(data, this._dataset.key());
+    } else {
+      dataElements = this.selection().data(data);
+    }
```

Comment:
Mainly curious: usually we have users re-animate their plot using `dataset.data(data)` even though that is hacky in my opinion.  After this change, would that idea still work?